### PR TITLE
e2e/service.go: remove same node nodeport test

### PR DESCRIPTION
**Pull request by aauren**
#### What type of PR is this?

/kind bug
/kind cleanup

#### What this PR does / why we need it:

Remove local pod -> local nodeport test from `Services [It] should fallback to local terminating endpoints when there are no ready endpoints with externalTrafficPolicy=Local` test.

How local external traffic policy influences behavior has been a hotly contested topic in the past. @danwinship created an issue several years ago for discussing exactly what the sig-network stance on this is here: https://github.com/kubernetes/kubernetes/issues/108526

In it, several sig-network members went back and forth, but ultimately decided that `local pod -> LB IP` and `local pod -> local NodePort` would remain exceptions from the "destination" interpretation of traffic policy.

However, in the case of `local pod -> local NodePort` the exception for this was primarily made because [sig-network didn't want to disrupt kube-proxy's existing functionality](https://github.com/kubernetes/kubernetes/issues/108526#issuecomment-1076845363). I think that this was ultimately the right call for kube-proxy, but I think that it still makes sense to allow other Kubernetes network providers to interpret this in a way that makes sense for them.

In kube-router's case, interpreting `local pod -> local NodePort` as external traffic not only keeps it consistent with the "destination" interpretation of traffic policy, but also keeps existing functionality in a decision similar to sig-network's WRT kube-proxy.

Given that:

* This e2e test is supposed to be testing traffic policy's termination logic and not the semantics of `local pod -> local NodePort` functionality
* That the current exception held by sig-network for `local pod -> local NodePort` is primarily held to keep kube-proxy consistent and not for it's ideological merits

I think that it would be beneficial to the community to remove this one line from the sig-network e2e battery of tests.

#### Special notes for your reviewer:

FYI @danwinship @aojea @andrewsykim 

Existing conversation on this can be found in Slack thread: https://kubernetes.slack.com/archives/C09QYUH5W/p1709310311449749

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

### CODE REVIEW

File: test/e2e/network/service.go [hunk 1]
```diff
@@ -3119,7 +3119,6 @@ var _ = common.SIGDescribe("Services", func() {
 
 			execHostnameTest(*pausePod0, nodePortAddress0, webserverPod0.Name)
 			execHostnameTest(*pausePod1, nodePortAddress0, webserverPod0.Name)
-			execHostnameTest(*pausePod1, nodePortAddress1, webserverPod0.Name)
```
Comment (aojea) [inline test/e2e/network/service.go:3122]: we need to update the comments above to explain what is going on, and also mention that this case is not tested with references to the discussions

Comment (aojea) [reply test/e2e/network/service.go:3122]: ok, so this assumes that  a pod serving in in nodeA should be reachable from a pod in nodeB that connectes to a nodePort in the same node .... 😄  a bit of a stretch I'm fine with this change, seems an artifact of kube-proxy implementation and at minimum debatable from a more fine interpretation

Comment (danwinship) [reply test/e2e/network/service.go:3122]: I need to do some more updates to https://github.com/kubernetes-sigs/kpng/blob/master/doc/service-proxy.md. It mentions the LB IP case but doesn't say anything about NodePorts+eTP:Local.

Anyway, I agree that this is testing an artifact of kube-proxy's implementation that I don't think we actually agreed was supposed to be the behavior for _all_ service proxy implementations. (Which means maybe we should make nftables _not_ do it that way?)

Note that we already have [a unit test for pod-to-eTP:Local-NodePort in iptables](https://github.com/kubernetes/kubernetes/blob/v1.29.0/pkg/proxy/iptables/proxier_test.go#L5582-L5596) that asserts the same behavior as seen here, so removing this test will not leave us in danger of regressing that.

Comment (aauren) [reply test/e2e/network/service.go:3122]: Updated the comments, let me know what you think.

**Comment by aojea** [comment]
/lgtm
/approve

https://github.com/kubernetes/kubernetes/pull/123622#discussion_r1509651847

**Pull request merged**